### PR TITLE
return 'sending' status for an event that is only locally redacted

### PR DESCRIPTION
**Pull request by bwindels**
Related to https://github.com/vector-im/riot-web/issues/10080

### CODE REVIEW

File: src/models/event.js [hunk 1]
```diff
@@ -877,16 +878,24 @@ utils.extend(module.exports.MatrixEvent.prototype, {
...
     /**
-     * Returns the status of the event, or the replacing event in case `makeReplace` has been called.
+     * Returns the status of any associated edit or redaction
+     * (not for reactions/annotations are their local echo doesn't affect the orignal event),
+     * or else the status of the event.
      *
      * @return {EventStatus}
      */
-    replacementOrOwnStatus() {
+    getAssociatedLocalEchoStatus() {
```
Comment (jryans) [inline src/models/event.js]: Maybe just `getAssociatedStatus`? `LocalEcho` seems a bit redundant.

File: src/models/event.js [hunk 2]
```diff
@@ -877,16 +878,24 @@ utils.extend(module.exports.MatrixEvent.prototype, {
...
 
     /**
-     * Returns the status of the event, or the replacing event in case `makeReplace` has been called.
+     * Returns the status of any associated edit or redaction
+     * (not for reactions/annotations are their local echo doesn't affect the orignal event),
+     * or else the status of the event.
      *
      * @return {EventStatus}
      */
-    replacementOrOwnStatus() {
```
[the remark this answers is not in the document]
Comment (bwindels) [reply src/models/event.js]: Indeed, it's relying on an unstable feature, so would not consider it a breaking change.

**Comment by bwindels** [comment]
Closing this in favor of https://github.com/matrix-org/matrix-js-sdk/pull/960 to make build in react sdk pass

File: src/models/event.js (hunk 1 above)
Comment (bwindels) [reply src/models/event.js]: fixed in #960

File: src/models/event.js [hunk 3]
```diff
@@ -877,16 +878,24 @@ utils.extend(module.exports.MatrixEvent.prototype, {
     },
 
     /**
-     * Returns the status of the event, or the replacing event in case `makeReplace` has been called.
+     * Returns the status of any associated edit or redaction
+     * (not for reactions/annotations are their local echo doesn't affect the orignal event),
```
[the remark this answers is not in the document]
Comment (bwindels) [reply src/models/event.js]: fixed in #960

**Pull request closed without merging**